packages/replay/src/util/handleRecordingEmit.ts

```diff
@@ -84,3 +93,38 @@ export function getHandleRecordingEmit(replay: ReplayContainer): RecordingEmitCa
     });
   };
 }
+
+/**
+ * Add an event to the event buffer.
+ * `isCheckout` is true if this is either the very first event, or an event triggered by `checkoutEveryNms`.
+ */
+export async function addSettingsEvent(replay: ReplayContainer, isCheckout?: boolean): Promise<void | null> {
```

Comment:
I would just make this `Promise<void>`, we don't need/care about the response?
Also no need to make this async. Actually, I wonder if we even need to return anything at all, or just make it `void` altogether?

Reply:
I think we should chain the return value of `addEvent` -- even though we don't use the return value, it's good to inform future folks that this function is async and is expected that it is async.